.gitignore

```diff
@@ -1 +1,34 @@
+# Numerous always-ignore extensions
+*.diff
+*.err
+*.orig
+*.log
+*.rej
+*.swo
+*.swp
+*.vi
+*~
+*.sass-cache
+node_modules/
+source/css/
+.tmp/
+package-lock.json
```

Comment:
`package-lock.json` should ALWAYS be present in the repo structure. We shouldn't ignore it.